data/src/main/java/org/apache/iceberg/data/avro/GenericReaders.java

```diff
@@ -128,11 +128,13 @@ protected Record reuseOrCreate(Object reuse) {
     }
 
     @Override
+    @SuppressWarnings("checkstyle:hiddenField")
```

Comment:
I know this rule is a bit controversial but I think it is better to follow it everywhere or ignore it globally.
In this particular case, it seems appropriate to follow and, maybe, give another name to the field. It is not really clear the difference between the field and passed object.

Reply:
You are right. I didn't look into it closely enough. I was thinking that baseline was mistakenly identifying this as a hidden field when non was present